### PR TITLE
Do not shade parquet-hadoop in delta-standalone

### PR DESCRIPTION
**Pull request by scottsand-db**
This PR has delta-standalone no longer shade `parquet-hadoop`. It then moves the `standalone-parquet` files, such as `ParquetSchemaConverter.java`, back into delta-standalone.

## Changes to pom.xml
We delete
```xml
<dependency>
    <groupId>io.delta</groupId>
    <artifactId>delta-standalone-parquet_2.12</artifactId>
    <version>0.5.0-SNAPSHOT</version>
</dependency>
```
and add 
```xml
<dependency>
    <groupId>org.apache.parquet</groupId>
    <artifactId>parquet-hadoop</artifactId>
    <version>1.12.0</version>
    <scope>provided</scope>
</dependency>
```

## Changes to delta-standalone (cosmetic) JAR
We remove
```
shadedelta/org/apache/commons/**
shadedelta/org/apache/yetus/**
shadedelta/org/apache/parquet/**
shadedelta/shaded/parquet/com/fasterxml/jackson/**
```

and add
```
io/delta/standalone/util/
```

These `zip`s contain
```
master_delta-hive-assembly-0.5.0-SNAPSHOT.jar
master_delta-standalone_2.12-0.5.0-SNAPSHOT.pom
master_hive_jar_depts.txt
master_standalone_jar_dependencies.txt
this_pr_delta-hive-assembly-0.5.0-SNAPSHOT.jar
this_pr_delta-standalone_2.12-0.5.0-SNAPSHOT.pom
this_pr_hive_jar_depts.txt
this_pr_standalone_jar_dependencies.txt
```

[master.zip](https://github.com/delta-io/connectors/files/9169349/master.zip)
[this_pr.zip](https://github.com/delta-io/connectors/files/9169352/this_pr.zip)

## Testing
Grzegorz Kołakowski was kind enough to perform the following testing.
image

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -495,8 +482,6 @@ lazy val standalone = (project in file("standalone"))
       ShadeRule.rename("org.json4s.**" -> "shadedelta.@0").inAll,
       ShadeRule.rename("com.github.mjakubowski84.parquet4s.**" -> "shadedelta.@0").inAll,
       ShadeRule.rename("org.apache.commons.pool.**" -> "shadedelta.@0").inAll,
-      ShadeRule.rename("org.apache.parquet.**" -> "shadedelta.@0").inAll,
-      ShadeRule.rename("shaded.parquet.**" -> "shadedelta.@0").inAll,
       ShadeRule.rename("org.apache.yetus.audience.**" -> "shadedelta.@0").inAll
```
Comment (scottsand-db) [inline build.sbt:485]: we don't need this anymore

**Comment by scottsand-db** [comment]
TODO: test hive. Probably need to add back `"org.apache.parquet" % "parquet-hadoop" % "1.12.0" % "provided",`.

**Comment by scottsand-db** [comment]
Tested this out on an EMR cluster and got

```
java.lang.NoClassDefFoundError: org/apache/parquet/schema/LogicalTypeAnnotation
  at shadedelta.com.github.mjakubowski84.parquet4s.LogicalTypes$.<init>(ParquetSchemaResolver.scala:107)
  at shadedelta.com.github.mjakubowski84.parquet4s.LogicalTypes$.<clinit>(ParquetSchemaResolver.scala)
  at shadedelta.com.github.mjakubowski84.parquet4s.SchemaDefs.$init$(ParquetSchemaResolver.scala:249)
  at shadedelta.com.github.mjakubowski84.parquet4s.ParquetSchemaResolver$.<init>(ParquetSchemaResolver.scala:33)
  at shadedelta.com.github.mjakubowski84.parquet4s.ParquetSchemaResolver$.<clinit>(ParquetSchemaResolver.scala)
  at io.delta.standalone.internal.Checkpoints$.writeCheckpoint(Checkpoints.scala:253)
  at io.delta.standalone.internal.Checkpoints.checkpoint(Checkpoints.scala:125)
  at io.delta.standalone.internal.Checkpoints.checkpoint$(Checkpoints.scala:121)
  at io.delta.standalone.internal.DeltaLogImpl.checkpoint(DeltaLogImpl.scala:42)
  at io.delta.standalone.internal.OptimisticTransactionImpl.postCommit(OptimisticTransactionImpl.scala:389)
  at io.delta.standalone.internal.OptimisticTransactionImpl.commit(OptimisticTransactionImpl.scala:155)
  at $anonfun$generateLargeCommitWithDSW$1(<console>:49)
  at $anonfun$generateLargeCommitWithDSW$1$adapted(<console>:45)
  at scala.collection.immutable.Range.foreach(Range.scala:158)
  at generateLargeCommitWithDSW(<console>:45)
  at $anonfun$res0$1(<console>:35)
  at scala.runtime.java8.JFunction1$mcVI$sp.apply(JFunction1$mcVI$sp.java:23)
  at scala.collection.immutable.List.foreach(List.scala:392)
  ... 58 elided
Caused by: java.lang.ClassNotFoundException: org.apache.parquet.schema.LogicalTypeAnnotation
  at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
  at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
  ... 76 more

```

**Comment by zsxwing** [comment]
This has been fixed by #406

**Pull request closed without merging**